engineyard/ey-cookbooks-stable-v5
# Do not call memcached from main recipes

## Description of your patch

Removes unnecessary dependency of the core recipes on memcached. Memcached being automatically installed on app instances was a legacy from the V4 stack.

## Recommended Release Notes

Removes unnecessary dependency of the core recipes on memcached.

## Estimated risk

Medium.

## Components involved

The following core recipes have been modified to remove calls to memcached:
- app
- collectd
- ey-base
- motd
- util

## Description of testing done

See QA instructions

## QA Instructions

A. Solo, new environment
1. Boot a new solo environment on the QA stack. 
2. Verify that chef runs without errors
3. Verify that the collectd memcached plugin is not installed
4. Verify that Performance Graphs on the EY Cloud dashboard is working
5. Enable memcached.
6. Verify that chef runs without errors.
7. Verify that the collectd memcached plugin is installed
8. Verify that Performance Graphs on the EY Cloud dashboard is working

B. Cluster, new environment
1. Boot a new cluster environment on the QA stack. Add a utility instance to the environment.
2. Verify that chef runs without errors
3. Verify that the collectd memcached plugin is not installed
4. Verify that Performance Graphs on the EY Cloud dashboard is working
5. Enable memcached.
6. Verify that chef runs without errors.
7. Verify that the collectd memcached plugin is installed on the memcached instance
8. Verify that the collectd memcached plugin is not installed on the other instances (app_master, db_master)
9. Verify that Performance Graphs on the EY Cloud dashboard is working

C. Solo, existing environment, upgrade scenario
1. Boot a new solo environment on the stable-v5 stack. Upgrade to the QA stack.
2. Verify that chef runs without errors
3. Verify that the collectd memcached plugin is not installed
4. Verify that Performance Graphs on the EY Cloud dashboard is working
5. Enable memcached.
6. Verify that chef runs without errors.
7. Verify that the collectd memcached plugin is installed
8. Verify that Performance Graphs on the EY Cloud dashboard is working

D. Cluster, existing environment, upgrade scenario
1. Boot a new cluster environment on the stable-v5 stack. Add a utility instance to the environment. Upgrade to the QA stack.
2. Verify that chef runs without errors
3. Verify that the collectd memcached plugin is not installed
4. Verify that Performance Graphs on the EY Cloud dashboard is working
5. Enable memcached.
6. Verify that chef runs without errors.
7. Verify that the collectd memcached plugin is installed on the memcached instance
8. Verify that the collectd memcached plugin is not installed on the other instances (app_master, db_master)
9. Verify that Performance Graphs on the EY Cloud dashboard is working
